src/Mvc/test/Mvc.FunctionalTests/TestingInfrastructureInheritanceTests.cs

```diff
@@ -80,6 +81,21 @@ public void TestingInfrastructure_GenericHost_HostShouldStopBeforeDispose()
             Assert.True(callbackCalled);
         }
 
+        [Fact]
+        public async Task TestingInfrastructure_GenericHost_HostDisposeAsync()
+        {
+            // Act
+            await using var factory = new CustomizedFactory<GenericHostWebSite.Startup>();
```

Comment:
Make sure this adds a service that is only IAsyncDisposable, that's the real test.